test/watcher_test.js

```diff
@@ -139,6 +162,65 @@ describe('Watcher', function() {
       expect(builderBuild).to.have.been.called;
     });
 
+    it('on change, rebuild is invoked and cancel is invoked from the build', function() {
+      const pluginsCalled = {};
+
+      const waitForPlugin1 = defer();
+      const waitForPlugin2 = defer();
+
+      class Plugin1 extends Plugin {
+        constructor(inputNodes, options) {
+          super(inputNodes || [], options);
+        }
+
```

Comment:
Why do we need this? It seems to me like we control the nodes passed, and we could just make sure to pass the right thing when constructing.
```suggestion
```